src/status_im/transport/inbox.cljs

```diff
@@ -0,0 +1,191 @@
+(ns status-im.transport.inbox
+  (:require [re-frame.core :as re-frame]
+            [status-im.native-module.core :as status]
+            [status-im.utils.handlers :as handlers]
+            [status-im.transport.utils :as web3.utils]
+            [taoensso.timbre :as log]))
+
+(defn- parse-json
+  ;; NOTE(dmitryn) Expects JSON response like:
+  ;; {"error": "msg"} or {"result": true}
+  [s]
+  (try
+    (let [res (-> s
+                  js/JSON.parse
+                  (js->clj :keywordize-keys true))]
+      ;; NOTE(dmitryn): AddPeer() may return {"error": ""}
+      ;; assuming empty error is a success response
+      ;; by transforming {"error": ""} to {:result true}
+      (if (and (:error res)
+               (= (:error res) ""))
+        {:result true}
+        res))
+    (catch :default e
+      {:error (.-message e)})))
+
+(defn- response-handler [error-fn success-fn]
+  (fn handle-response
+    ([response]
+     (let [{:keys [error result]} (parse-json response)]
+       (handle-response error result)))
+    ([error result]
+     (if error
+       (error-fn error)
+       (success-fn result)))))
+
+(defn add-peer [enode success-fn error-fn]
```

Comment:
Basic docstring

Reply:
Similar to below public fns. I realize a lot of this is technical debt from offline inboxing swarm, but if are going to do this massive one change let's get it right and document core transport functionality in code :) Bit of a time investment but will be valuable for a long time going forward, especially for general individual contributors